src/TodoList.coffee

```diff
@@ -0,0 +1,94 @@
+define 'TodoList',
+[
+    './ViewInterface.coffee',
+    './ObservableCollection.coffee'
+],
+(ViewInterface, ObservableCollection) ->
+    class TodoList extends ViewInterface
+
+        observableItems = new ObservableCollection()
+        todoListSelector = document.querySelector('.todo-list')
+        formSelector = document.querySelector('.todo-list-form')
+
+        init: ->
+            _render()
+            _initDOMListeners()
+            _initWebsocketConnection()
+        _initWebsocketConnection = ->
+            socket = new WebSocket('ws://localhost:8080')
+            socket.onopen = ->
+                socket.send(JSON.stringify({ type: 'GET_DATA' }))
+                observableItems.onChange((data) ->
+                    socket.send(JSON.stringify({ type: 'UPDATE_DATA', payload: data }))
+                )
+            socket.onmessage = (event) ->
+                data = JSON.parse(event.data)
+                observableItems.mutateSilent(data)
+                _render()
+            socket.onclose = (event) ->
+                if (event.wasClean)
+                    console.log('Соединение закрыто чисто')
+                else
+                    console.log('Обрыв соединения')
+                console.log('Код: ' + event.code + ' причина: ' + event.reason)
+            socket.onerror = (error) ->
+                console.log("Ошибка " + error.message)
+        _initDOMListeners = ->
+            formSelector.addEventListener('submit', _addItem.bind(@))
+            document.addEventListener('click', _deleteItem.bind(@))
+            document.addEventListener('click', _editItem.bind(@))
+            todoListSelector.addEventListener('DOMNodeRemoved', _onDestroy.bind(@))
+        _onDestroy = ->
+            formSelector.removeEventListener('submit', _addItem.bind(@))
+            document.removeEventListener('click', _deleteItem.bind(@))
+            document.removeEventListener('click', _editItem.bind(@))
+        _editItem = (e) ->
+            { target } = e
+            if (target.classList.contains('edit-btn'))
+                text = target.parentElement.dataset.key
+                spanSelector = target.parentElement.querySelector('span.todo-item-text')
```

Comment:
It is good practice to use dollar sign for selectors. Like $spanSelector

Reply:
done